### ISSUE TITLE
List of travelers who have traveled a system

### ISSUE THREAD

**Issue by michihdeu** [TravelMapping/Web#97]
A ranking like we had: http://cmap.m-plex.com/stat/travsystem.php?sys=deua.

Does a ranking like this already exist?
How is it accessable from the start page?

What's about a ranking of a selected region?

**Comment by michihdeu** [comment]
Does a ranking exist or is it a feature?

**Comment by michihdeu** [comment]
http://web.archive.org/web/20160422033800/http://cmap.m-plex.com/stat/travsystem.php?sys=deua

A user ranking per system does not yet exist (only per region) but it should be here: http://travelmapping.net/user/system.php?sys=deua

**Comment by yakra** [comment]
I like the idea of breaking out the user ranking tables into a separate PHP page, as done on CHM.

**Comment by jteresco** [comment]
This is something I should add soon, while I'm in PHP/SQL thinking mode.  I don't know why it was never there originally.

**Comment by michihdeu** [comment]
@jteresco is this still on your list?

**Comment by jteresco** [comment]
Maybe.  I'm in class prep mode now, so not much TM development time.

**Comment by yakra** [comment]
https://tmstage.teresco.org/user/region.php?u=formulanone&rg=ME
https://tmstage.teresco.org/user/system.php?u=formulanone&sys=usame
https://tmstage.teresco.org/user/system.php?u=formulanone&sys=usame&rg=ME

Can this be closed now?

**Comment by michihdeu** [comment]
thx, closing

**Comment by jteresco** [comment]
Closing so soon?  It's not quite been 5 years... 

Thanks @yakra for making it happen.

**Comment by michihdeu** [comment]
@yakra

- [x]  when clicking on a user in the ranking table of region.php (rg specific in url), the page is loaded for the region with the selected user.
- [x]  when clicking on a user in the ranking table of system.php (sys specific in url), the page is loaded for the system with the selected user.
- [ ]  when clicking on a user in the ranking table of system.php (sys **+ rg** specific in url), the page is loaded for the system with the selected user **but the region filter gets lost!**

Can you fix it short-term or should I open a new issue?

**Comment by yakra** [comment]
A fix is live on lab2. Sent a PM on the forum.

**Comment by michihdeu** [comment]
> A fix is live on lab2. Sent a PM on the forum.

Looks good! 👍

**Comment by michihdeu** [comment]
@yakra The little "back to user page" link next to "View larger map" doesn't work on lab2.
It's fine on the producation server and on tmstage

**Comment by yakra** [comment]
I think this may be an issue with cookies.
@jteresco, are cookies set up to only do their magic via https now?

**Comment by michihdeu** [comment]
I just saw it when clicking around. Everything should be fine if you have not changed anything on that with the latest fix. We should double-check it on tmstage though.

**Comment by jteresco** [comment]
I don't know of anything that restricts cookies to https only, but I suppose someone's browser might disallow it.

**Comment by michihdeu** [comment]
> The little "back to user page" link next to "View larger map" doesn't work on lab2

It still works on tmstage with the latest fix. It must be a lab2-specific issue.

region.php has a nice "Scroll down to see statistical reports below the map" text next to that very "back to user page" button.
system.php misses the text.

region,php's "back to user page" is just called "User Page" on system.php.
The link is superfluous since we introcuded the "User Page" button in the header.

**Comment by yakra** [comment]
@michihdeu [wrote](https://github.com/TravelMapping/Web/issues/97#issuecomment-865222245):
> It still works on tmstage with the latest fix. It must be a lab2-specific issue.

Lab2 specific. The code that generates the link hasn't changed. Hover over the link and see that no `u=` argument is passed to the page, meaning it's retrieved from cookie info once we go over to that page.

@jteresco [wrote](https://github.com/TravelMapping/Web/issues/97#issuecomment-865217190):
> I don't know of anything that restricts cookies to https only, but I suppose someone's browser might disallow it.

ISTR cookies stopped working on lab2 around the time the changeover to HTTPS was made. Never looked into it too closely, just said "Oh, that's broken now" and went on my merry way.

Cookies work in both Chromium & Firefox with noreaster.
Work in neither with lab2, though IIRC they used to. Haven't changed anything WRT browser config. Also, @michihdeu experienced the same problem so I doubt it's a browser-side thing.

Once #654 is merged, I'll do a `git bisect` on lab2 and find when things broke.

Not opening up a new issue for this, as it might not even be anything in TravelMapping's code. Might just be something I need to reconfig in Apache?

**Comment by yakra** [comment]
Weird. The process of trying to figure out where stuff broke... fixed it. Even with the most recent Web commit, where stuff was behaving badly before.
@michihdeu, do the links to the main user page on lab2 work now; does the page display properly?

**Comment by michihdeu** [comment]
It's live on the production server. Closing.
